packages/commons-server/src/libs/server/server.ts

```diff
@@ -214,26 +230,31 @@ export class MockoonServer extends (EventEmitter as new () => TypedEmitter<Serve
     app.disable('etag');
 
     this.generateDatabuckets(this.environment);
+    app.use(this.parseBody); // This middleware is required to parse the body for createAdminEndpoint requests
 
     // admin endpoint must be created before all other routes to avoid conflicts
-    createAdminEndpoint(app, {
-      statePurgeCallback: () => {
-        // reset request numbers
-        Object.keys(this.requestNumbers).forEach((routeUUID) => {
-          this.requestNumbers[routeUUID] = 1;
-        });
+    createAdminEndpoint(
+      app,
+      {
+        statePurgeCallback: () => {
+          // reset request numbers
+          Object.keys(this.requestNumbers).forEach((routeUUID) => {
+            this.requestNumbers[routeUUID] = 1;
+          });
 
-        // reset databuckets
-        this.processedDatabuckets = [];
-        this.generateDatabuckets(this.environment);
-      }
-    });
+          // reset databuckets
+          this.processedDatabuckets = [];
+          this.generateDatabuckets(this.environment);
+        }
```

Comment:
Could you add a call to `this.purgeGlobalVariables` here, so the full state is purged?